### DISCUSSION TITLE
SMI measure faults

### DISCUSSION THREAD

**Discussion by z3r0tnd** [arendst/Tasmota#18756] [Q&A]
Hi there,

I've bought a Hichi ESP01s to read out my smartmeter.
Under https://tasmota.github.io/docs/Smart-Meter-Interface/ there is a already made script for my smartmeter 
which worked at the beginning without any problems.

Script:
`>D`
`>B`
`->sensor53 r`
`>M 1`
`+1,3,s,16,9600,SML`
`1,77070100600100ff@#,Server ID,,server_id,0`
`1,77070100020800ff@1000,Export (Total),kWh,export_total_kwh,4`
`1,77070100010800ff@1000,Consumption (Total),kWh,total_kwh,4`
`1,77070100100700ff@1,Consumption (Current),W,curr_w,0`
`1,77070100200700ff@1,Voltage L1,V,volt_p1,1`
`1,77070100340700ff@1,Voltage L2,V,volt_p2,1`
`1,77070100480700ff@1,Voltage L3,V,volt_p3,1`
`1,770701001f0700ff@1,Amperage L1,A,amp_p1,1`
`1,77070100330700ff@1,Amperage L2,A,amp_p2,1`
`1,77070100470700ff@1,Amperage L3,A,amp_p3,1`
`1,77070100510701ff@1,Phase angle U-L2/U-L1,deg,phase_angle_l2_l1,1`
`1,77070100510702ff@1,Phase angle U-L3/U-L1,deg,phase_angle_l3_l1,1`
`1,77070100510704ff@1,Phase angle I-L1/U-L1,deg,phase_angle_p1,1`
`1,7707010051070fff@1,Phase angle I-L2/U-L2,deg,phase_angle_p2,1`
`1,7707010051071aff@1,Phase angle I-L3/U-L3,deg,phase_angle_p3,1`
`1,770701000e0700ff@1,Frequency,Hz,freq,0`
`#`

It takes around 1 1/2 month until this measurements start:
![sec](https://github.com/arendst/Tasmota/assets/20965118/b600b1f3-e8df-4a6b-9249-318a339e1bfa)

Normaly it looks like:
![ok](https://github.com/arendst/Tasmota/assets/20965118/5c442185-10d9-4a1e-80ad-8c20109eeedd)

Console Example:
`0:07:16.414 MQT: tele/tasmota_4EEBC6/STATE = {"Time":"2023-05-30T20:07:16","Uptime":"0T00:04:38","UptimeSec":278,"Heap":17,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"Schlange","BSSId":"B0:F2:08:CA:C8:B9","Channel":3,"Mode":"11n","RSSI":56,"Signal":-72,"LinkCount":1,"Downtime":"0T00:00:03"}}`
`20:07:16.422 MQT: tele/tasmota_4EEBC6/SENSOR = {"Time":"2023-05-30T20:07:16","SML":{"server_id":"0123456","export_total_kwh":999.9990,"total_kwh":4624.3191,"curr_w":562,"volt_p1":999999.0,"volt_p2":999999.0,"volt_p3":999999.0,"amp_p1":1.2,"amp_p2":1.7,"amp_p3":0.3,"phase_angle_l2_l1":120.0,"phase_angle_l3_l1":999999.0,"phase_angle_p1":325.0,"phase_angle_p2":329.0,"phase_angle_p3":270.0,"freq":999999}}`
`20:07:46.413 MQT: tele/tasmota_4EEBC6/STATE = {"Time":"2023-05-30T20:07:46","Uptime":"0T00:05:08","UptimeSec":308,"Heap":17,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"Schlange","BSSId":"B0:F2:08:CA:C8:B9","Channel":3,"Mode":"11n","RSSI":58,"Signal":-71,"LinkCount":1,"Downtime":"0T00:00:03"}}`
`20:07:46.421 MQT: tele/tasmota_4EEBC6/SENSOR = {"Time":"2023-05-30T20:07:46","SML":{"server_id":"0123456","export_total_kwh":82.5518,"total_kwh":4624.3240,"curr_w":540,"volt_p1":230.3,"volt_p2":231.1,"volt_p3":230.3,"amp_p1":1.2,"amp_p2":1.6,"amp_p3":0.3,"phase_angle_l2_l1":120.0,"phase_angle_l3_l1":239.0,"phase_angle_p1":325.0,"phase_angle_p2":327.0,"phase_angle_p3":270.0,"freq":50}}`

I don't understand why it is measuring sometimes such wrong values.

**Comment by hagenbuch** [comment]
You might look into the "raw data" of the protocol by turning logging on:

    Sensor53 D1

Don't forget to turn it off (D0) for it won't scrape data while dumping the data.

**Comment by z3r0tnd** [comment]
Hi there,

I had some contact to the seller. 
One hint is currently helping me.
This is not concerning tasmota Software but is working for me.
I only put two papers between the hichi and smartmeter.
Lets see how long this will Help.
